command/operator_raft_autopilot_set_config.go

```diff
@@ -138,6 +155,10 @@ func (c *OperatorRaftAutopilotSetConfigCommand) Run(args []string) int {
 		data["disable_upgrade_migration"] = c.flagDisableUpgradeMigration.Get()
 	}
 
+	if c.flagDRToken != "" {
+		client.SetToken(c.flagDRToken)
```

Comment:
I'm curious about this pattern of using `client.SetToken()` for passing along the DR token. If this works, why not use it for the state and get config endpoints too, instead of creating special methods for them?

Reply:
I haven't verified it, but the fact that DR operation tokens are their own parameters to API calls makes me dubious that this works.

Reply:
I had a similar thought/questions.  And from looking into the code and the original change to do it that way (https://github.com/hashicorp/vault/pull/10856/) It still wasn't very clear to me.  Perhaps @vishalnayak might be able to provide some light, since they made the original change (albeit a while back).

Reply:
It's not clear to me that this is a valid pattern. I would prefer if this was done in a similar style to how it's done for other raft commands that support a `-dr-token` flag, e.g. [list-peers](https://github.com/hashicorp/vault/blob/main/command/operator_raft_listpeers.go#L88-L91).

Reply:
I understand what you are saying now.  I will work on that 👍

Reply:
So I looked at this a bit more and it looks like this is where we set a DR token (if passed through), or default to the ClientToken. https://github.com/hashicorp/vault-enterprise/blob/main/vault/replication_api_ent.go#L803-L808.  I think you are correct that it is probably better to set this token explicitly, rather than just set a client token (since this looks like it is already built in). I went ahead and pushed that commit for you to review.